kernel/Utilities/GlobalIndexing.h

```diff
@@ -493,6 +423,78 @@ class GlobalIndexing {
                    globalIndex < blockStart_ + numberOfBasisFunctionsInBlock_;
         }
     };
+
+    /**
+     * Visitor to access the *Offsets_ members (e.g. elementOffsets_) of the
+     * Offsets class. If there is no offset entry available for the MeshEntity
+     * then it will be created (and initialized to the default 0).
+     */
+    struct OffsetVisitor : public Base::MeshEntityVisitor<> {
+        OffsetVisitor(Offsets& offsets)
+            : visitOffsets_(offsets), entityOffset_(nullptr){};
+
+        void visit(Base::Element& element) final {
+            entityOffset_ = &visitOffsets_.elementOffsets_[element.getID()];
+        }
+        void visit(Base::Face& face) final {
+            entityOffset_ = &visitOffsets_.faceOffsets_[face.getID()];
+        }
+        void visit(Base::Edge& edge) final {
+            entityOffset_ = &visitOffsets_.edgeOffsets_[edge.getID()];
+        }
+        void visit(Base::Node& node) final {
+            entityOffset_ = &visitOffsets_.nodeOffsets_[node.getID()];
+        }
+
+        Offsets& visitOffsets_;
```

Comment:
could we have a `const Reference`

Reply:
(see also previous) That would prevent editing the offset.

Reply:
Yeah makes sense